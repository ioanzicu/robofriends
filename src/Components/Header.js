import React, { memo } from 'react'

const Header = ({ title }) => {
  return (
    <h1 id='title' className='f1'>
      {title}
    </h1>
  )
}

const headerPropsAreEqual = (prevProps, nextProps) =>
  prevProps.title === nextProps.title

export const MemoizedHeader = memo(Header, headerPropsAreEqual)
