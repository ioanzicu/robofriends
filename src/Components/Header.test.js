import React from 'react'
import { shallow } from 'enzyme'
import { MemoizedHeader } from './Header'

it('expect to render Header component', () => {
  const headerComponent = shallow(<MemoizedHeader />)
  expect(headerComponent.debug()).toMatchSnapshot()
})
