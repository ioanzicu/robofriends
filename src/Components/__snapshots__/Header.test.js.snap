// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header component 1`] = `"<h1 id=\\"title\\" className=\\"f1\\" />"`;
